include/aie/Dialect/AIEVec/IR/AIEVecOps.td

```diff
@@ -667,4 +667,29 @@ def AIEVec_MaxOp:
   let assemblyFormat = "$lhs `,` $rhs attr-dict `:` type($result)";
   let hasVerifier = 0;
 }
+
+def AIEVec_CmpOp:
+  AIEVec_Op<"cmp", [
+    Pure,
+    AllTypesMatch<["lhs", "rhs"]>
+  ]>,
+  Arguments<(ins AnyVector:$lhs, AnyVector:$rhs, DefaultValuedStrAttr<StrAttr, "">:$pred)>,
```

Comment:
We are abusing the use of attributes. If it must always be present, the predicate should be another operand, not an attribute. You probably use the same type constraints that `arith.cmp` uses for this.